modules/server/src/network/resolvers/aggregations.ts

```diff
@@ -1,239 +1,85 @@
-
-	// ensure requested field is available to query on node
-	const fieldsToRequest = Object.keys(requestedFields).reduce((acc, requestedFieldKey) => {
-		const field = requestedFields[requestedFieldKey];
-		if (availableFields.some((field) => field.name === requestedFieldKey)) {
-			return { ...acc, [requestedFieldKey]: field };
-		} else {
-			return acc;
-		}
-	}, {});
-
-	const gqlString = createNodeQueryString(documentName, fieldsToRequest);
-
-	/*
-	 * convert string to AST object
-	 * not needed if gqlString is formatted correctly but this acts as a validity check
-	 */
-	try {
-		const gqlQuery = gql`
-			${gqlString}
-		`;
-		return gqlQuery;
-	} catch (err) {
-		console.error('invalid gql', err);
-		return undefined;
-	}
+import { RequestedFieldsMap } from '../utils/gql';
+import { fetchData } from './fetch';
+import { createNetworkQuery } from './query';
+
+export const CONNECTION_STATUS = {
+	OK: 'OK',
+	ERROR: 'ERROR',
+} as const;
+
+export type NetworkNode = {
+	name: string;
+	hits: number;
+	status: keyof typeof CONNECTION_STATUS;
+	errors: string;
+	aggregations: { name: string; type: string }[];
 };
 
-type SuccessResponse = { [k: string]: { hits: Hits; aggregations: AllAggregations } };
+type SuccessResponse = Record<string, { hits: Hits; aggregations: AllAggregationsMap }>;
 
 /**
  * Query each remote connection
  *
  * @param queries - Query for each remote connection
- * @param requestedAggregationFields
- * @returns
+ * @param requestedAggregationFields - Fields requested
+ * @returns Resolved aggregation and node info
  */
 export const aggregationPipeline = async (
 	configs: NodeConfig[],
 	requestedAggregationFields: RequestedFieldsMap,
-	queryVariables: QueryVariables,
+	queryVariables: AggregationsQueryVariables,
 ) => {
 	const nodeInfo: NetworkNode[] = [];
 
 	const totalAgg = new AggregationAccumulator(requestedAggregationFields);
 
-	const aggregationResultPromises = configs.map(async (config) => {
-		const gqlQuery = createNetworkQuery(config, requestedAggregationFields);
-		const response = gqlQuery
-			? await fetchData<SuccessResponse>({
-					url: config.graphqlUrl,
-					gqlQuery,
-					queryVariables,
-			  })
-			: failure(CONNECTION_STATUS.ERROR, 'Invalid GQL query');
-
-		const nodeName = config.displayName;
-
-		if (isSuccess(response)) {
-			const documentName = config.documentName;
-			const responseData = response.data[documentName];
-			const aggregations = responseData?.aggregations || {};
-			const hits = responseData?.hits || { total: 0 };
-
-			totalAgg.resolve({ aggregations, hits });
-
-			nodeInfo.push({
-				name: nodeName,
-				hits: hits.total,
-				status: CONNECTION_STATUS.OK,
-				errors: '',
-				aggregations: config.aggregations,
-			});
-		} else {
-			nodeInfo.push({
-				name: nodeName,
-				hits: 0,
-				status: CONNECTION_STATUS.ERROR,
-				errors: response?.message || 'Error',
-				aggregations: config.aggregations,
-			});
-		}
-	});
+	await Promise.allSettled(
```

Comment:
TIL!!!

I do wonder how the community is not worried about having so many ambiguously synonymous words in the implementation
`resolved` vs `settled` vs `fulfilled`. it's all getting a bit wonky in my head now

Reply:
might be more confusing... but my understanding:
resolved = success
fulfilled = settled = success | failure

Just so happens that the Promise api uses the word "settled" instead of "fulfilled"